test/integration/transaction_test.exs

```diff
@@ -298,26 +298,45 @@ defmodule Xander.Integration.TransactionTest do
         end
       end)
 
-    # Now submit all transactions in sequence
-    Enum.each(signed_transactions, fn {index, tx_cbor, tx_signed} ->
-      case Xander.Transaction.send(tx_cbor) do
-        {:ok, tx_hash} ->
-          IO.puts("Successfully submitted transaction #{index} with Xander: #{tx_hash}")
-          assert :accepted == tx_hash
-
-        {:error, %{type: :refused}} ->
-          IO.puts(
-            "Handshake refused by node for transaction #{index}. Falling back to cardano-cli."
-          )
-
-          assert false
-
-        {:error, error} ->
-          IO.puts("Failed to submit transaction #{index} with Xander: #{error}")
-          # In CI, we'll just pass the test since we can't actually submit transactions
-          IO.puts("Skipping actual transaction submission check in CI environment")
-          assert true
-      end
-    end)
+    # Submit all transactions in parallel and collect responses
+    tasks =
+      Enum.map(signed_transactions, fn {index, tx_cbor, tx_signed} ->
+        Task.async(fn ->
+          case :gen_statem.call(Xander.Transaction, {:request, :send_tx, tx_cbor}) do
```

Comment:
Any reason why the code here needs to be different from `Xander.Transaction.send(tx_cbor)` ?

Reply:
nope; good call!